java/core/page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/support/capabilities/WebDriverCapabilitiesParser.java

```diff
@@ -0,0 +1,121 @@
+package ru.sbtqa.tag.pagefactory.support.capabilities;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import ru.sbtqa.tag.pagefactory.drivers.TagWebDriver;
+import static ru.sbtqa.tag.pagefactory.support.BrowserType.CHROME;
+import ru.sbtqa.tag.qautils.properties.Props;
+
+public class WebDriverCapabilitiesParser implements CapabilitiesParser {
+
+    // prefix is a 'webdriver.ie.capability.' or 'webdriver.*.capability.' part
+    private static final String CAPABILITY_WITH_PREFIX_REGEX = "webdriver.(" + TagWebDriver.getBrowserName() + "|\\*).capability.(.*)";
+
+    private final DesiredCapabilities capabilities = new DesiredCapabilities();
+    private final Map<String, Object> chromeOptions = new HashMap<>();
+
+    @Override
+    public DesiredCapabilities parse() {
+        Set<String> properties = Props.getProps().stringPropertyNames();
+        List<String> capabilitiesWithPrefix = getCapabilitiesWithPrefix(properties, CAPABILITY_WITH_PREFIX_REGEX);
+
+        for (String capabilityWithPrefix : capabilitiesWithPrefix) {
+
+            String capabilityName = cutPrefix(capabilityWithPrefix);
+            String capabilityValue = Props.get(capabilityWithPrefix);
+
+            switch (TagWebDriver.getBrowserName()) {
+                case CHROME:
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') End this switch case with an unconditional break, return or throw statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS128)